### PR TITLE
Revamping the Schema Documentation

### PR DESCRIPTION
**Pull request by jenningsanderson**
View the proposed changes here: https://dfhx9f55j8eg5.cloudfront.net/pr/102

1. Bumps to Docusaurus 3.1.0, which fixes the security vulnerabilities
2. Changes sidebar structure to address "common" elements, such as the new names schema.

Broadly, this takes the documentation from the previous "We are introducing the Overture schema, it will be like this" to "This is what the schema is"

### CODE REVIEW

**Comment by vcschapp** [comment]
I created a GitHub issue to track this work. My apologies for creating it after the fact.

https://github.com/OvertureMaps/schema-wg/issues/251

I put somewhat stringent "acceptance criteria" into that issue... Let me know if they make sense:

> 1. [ ] MUST add a sub-page structure into the documentation to cover "common" schema items.
> 2. [ ] MUST refactor the Transportation theme's [Scoped And Rule-Based Properties](https://docs.overturemaps.org/themes/transportation/scoping-rules) to lift it up to fit within the "common" schema sub-pages.
> 3. [ ]  MUST document the new names schema within a sub-page in the "common" schema area.

If these ACs make sense then I think this change only partially accomplishes item (3) but leaves item (1) and (2) incomplete...

-----

What if we structured things a bit more like this:

 > - Overview
>     - Introduction ...
>     - Feature Model
>         - Data Formats: GeoJSON and GeoParquet (no content currently)
>         - Names
>         - Scoped and Rule-Based Properties

Then:

1. The content of "Introduction" would mention that the Overture schema uses a simple feature model and that text would link out to the "Feature Model" page.
2. The content of the "Feature Model" page would have headings about:
    - Simple Features and why we use them
    - Common Feature Properties
        - This would list all the common properties maybe in a table. The entry for names would link to the Names page.
    - Common Concepts
        - This would list all the common concepts we can think of. The only concept I can currently think of is the Scoped and Rule-based properties concept, which would link out to that page.
    - Data Formats. This would have a sentence or two about canonical GeoJSON and GeoParquet and would link out to the sub-page for that item, which will be empty and listed as "under construction".

File: docusaurus/package.json [hunk 1]
```diff
@@ -17,19 +17,20 @@
...
+    "prism-react-renderer": "^2.1.0",
     "raw-loader": "^4.0.2",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
   },
   "devDependencies": {
-    "@docusaurus/module-type-aliases": "2.4.0",
+    "@docusaurus/module-type-aliases": "^3.1.0",
```
Comment (vcschapp) [inline docusaurus/package.json:32]: Big bounce! good to see those security issues going away.

**Review by vcschapp — COMMENTED** [review]
Left thoughts on the docs content.

**Comment by jenningsanderson** [comment]
@danabauer — We're considering some big updates to the structure of the schema documentation here. What are your thoughts?

File: docusaurus/docs/overview/feature-model/names.mdx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+Common name variations are translations of a name.
+
+```yaml
+names:
+  common:
+    es: La Casa Blanca
+```
+
+#### `rules`
+Additional variations such as _official_, _alternate_, or _short_ name variations exist as `rules`.
```
Comment (vcschapp) [inline docusaurus/docs/overview/feature-model/names.mdx:36]: 1. [ ] Ideally we should explain the link between `rules` and "Scoped and Rule-Based Properties".
2. [ ] Ideally we should give some further explanation of how to build up names rules and what your options are, *e.g.* language, side, LR, *etc.*
3. [ ] I think that the "rule evaluation algorithm" described on the scoped and rule-based properties page may need to be massaged to explain how to resolve name rules. I'm not 100% sure yet, but we should think about it.

File: docusaurus/docs/overview/feature-model/scoping-rules.mdx [hunk 3]
```diff
@@ -1,3 +1,8 @@
+---
+title: Scoping Rules
```
Comment (vcschapp) [inline docusaurus/docs/overview/feature-model/scoping-rules.mdx:2]: We maybe should change the title to "Scoping *and* rules"...

Technically, *scoping* and *rules* are two different concepts.

1. All rule-based properties involve scoping.
1. But a property can be scoped, without being rule-based.

File: docusaurus/docs/overview/feature-model/scoping-rules.mdx [hunk 4]
```diff
@@ -21,27 +26,27 @@ vary according to the day of the week.
...
 
 ### Geometric scoping (linear referencing)
 
 <Tabs>
   <TabItem value="description" label="Description" default>
 
 The geometric scoping property `at` limits the scope of its parent value to a
-position or range of positions along a Segment's geometry. When the parent value
+position or range of positions along a segment's geometry. When the parent value
 is a rule object, the rule only matches the position or range of positions
```
Comment (vcschapp) [inline docusaurus/docs/overview/feature-model/scoping-rules.mdx:50]: This paragraph highlights the fact that the page content was originally written from an *exclusively* transportation focus, and probably should be massaged to be more generically-focused.

The problem with this paragraph is it assumes "your brain context = transportation" so it starts talking about segments, which are a transportation context. Maybe in this particular example, what is meant is "linear features"?

Probably the whole page needs to be audited carefully from that perspective.

**Review by vcschapp — COMMENTED** [review]
Thanks @jenningsanderson and @danabauer!

I believe some changes should probably be done *after this PR* to smooth off some rough edges - some example comments are flagged within. But it's my preference to get this one merged and then maybe tackle those changes later, to avoid delaying the important changes.

@danabauer Would it be possible to add some of the comments to your TODO list? If so, I would suggest we approve and merge this one and just let the refinements come as they're ready.

**Comment by jenningsanderson** [comment]
Nice @vcschapp — I agree. Maybe we could get this PR into the minimum: "we're happy if this goes live with the next data release" state, meaning it has: 
- [x] Better intro
- [ ] Describes the common `name` schema
- [ ] Refreshes GERS pages a bit? 

at the minimum, and then we merge it into `dev` (so it's ready to get to pushed to `main`) — and then we can open a new PR with incremental changes?

**Comment by danabauer** [comment]
I'll fix this!

File: docusaurus/docs/overview/feature-model/index.mdx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    "extBar": "Me too!",
+    "theme": "buildings",
+    "type": "building",
+    "version": 1,
+    "level": 1,
+    "updateTime": "2023-06-06T10:30:00-08:00",
+    "height": 21.34,
+    "numFloors": 4,
+    "class": "civic",
+    "names": {
```
Comment (jwass) [inline docusaurus/docs/overview/feature-model/index.mdx:51]: This needs to be updated to latest names schema

File: docusaurus/docs/overview/feature-model/index.mdx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      "common": [
+        {
+          "value": "Springfield Town Hall",
+          "language": "local"
+        }
+      ]
+    },
```
Comment (jwass) [inline docusaurus/docs/overview/feature-model/index.mdx:58]:
```
"names": {
  "primary": "Springfield Town Hall"
},
```

**Review by jwass — APPROVED** [review]
Approved subject to recommended change

**Comment by vcschapp** [comment]
Reminder: @TristanDiet-TomTom and @DavidKarlas committed to reviewing this by no later than tomorrow. We also agreed that a "late merge" to `dev` past the freeze date is acceptable here because it is only docs.

**Comment by danabauer** [comment]
> We also agreed that a "late merge" to `dev` past the freeze date is acceptable 

That's great @vcschapp. What is the freeze date?

File: docusaurus/docs/gers/gers.mdx [hunk 7]
```diff
@@ -3,48 +3,72 @@ title: Global Entity Reference System
...
+## Components of a GERS ID
 
-1. An ever-growing set of _entities_ that are a shared reference for a `thing` in the world, where a `thing` could be a segment of road, a city, a store, a building, etc. Multiple _features_ in the Overture corpus can and will share the same GERS ID if they are representing the same `thing`.
+GERS generates IDs in a consistent format:
 
-2.  GERS IDs are stable (with a reasonable tolerance of error). Across multiple versions of Overture data, efforts will be taken to ensure the mapping of a real-world `thing` to a GERS ID remains consistent. When stability is not possible, traceability will be provided. Examples:
-    - A single road segment is bisected by a new road and becomes 2 road segments: 1 GERS ID → 2 New GERS IDs
-    - 1 large building footprint on the map is determined to be 4 smaller buildings when a higher resolution dataset becomes available: 1 GERS ID → 4 new GERS IDs
-    - A building is shifted 10m west when higher resolution imagery is made available: GERS ID is preserved for that feature.
+`H3 tile address` +
```
Comment (jenningsanderson) [inline docusaurus/docs/gers/gers.mdx:19]: @danabauer — I changed the order of these: h3 tile is first

File: docusaurus/docs/overview/feature-model/index.mdx [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+Overture data is described by the [_simple feature model_](https://www.ogc.org/standard/sfa/) specified by the Open Geospatial Consortium. This means that each Overture _entity_ is represented as a single [GeoJSON](https://geojson.org/) feature: a geometric object with associated attributes and properties. For example, an _entity_ from the Overture buildings data theme might be represented in this way:
+
+```json
+{
+  "id": "08ba0085884a4fff0200ab2e3ad3f37c",
+  "type": "Feature",
+  "geometry": {
+    "type": "Polygon",
+    "coordinates": [
```
Comment (jenningsanderson) [inline docusaurus/docs/overview/feature-model/index.mdx:15]: @danabauer — I took out some of the newlines here to make it more readable; do you think this is OK? Do you think we could even go to just 1 line for the coordinates?

**Comment by vcschapp** [comment]
> > We also agreed that a "late merge" to `dev` past the freeze date is acceptable
> 
> That's great @vcschapp. What is the freeze date?

Today technically!

Our current plan (until we think of a better one) is to freeze merges to the schema's `dev` branch two weeks before the release and unfreeze them on the day the release goes out.

File: docusaurus/docs/overview/feature-model/index.mdx (hunk 8 above)
Comment (danabauer) [reply docusaurus/docs/overview/feature-model/index.mdx:15]: looks good! No need to go to one line

**Review by TristanDiet-TomTom — APPROVED** [review]
Approving, good step forward.

But fully agreeing with the need to restructure afterward.
Names and mechanisms topics are currently a bit floating as children of the introduction and is a mixture of mechanism and feature content pages.
Additionally, looking at it from the perspective of a newcomer looking for an initial narrative in the schema: the current TOC goes:
1. Introduction (~basic welcome)
1.1 Mechanisms/properties that work across Layers (~advanced/expert from schema perspective
2. Layers (~basic from schema perspective)

I would group 1.1 and 2. in one Feature Model bucket with the 1.1 as the last sibling of the 2. topics. This 1.1 subchapter would explain a general introduction on layer crossing properties, the subpages would tackle the respective feature topic pages for names, gers identifier,....
Above is just an initial line of thought, but I would, post-PR, happily discuss additional optimizations.

**Comment by vcschapp** [comment]
Just missing approval from @DavidKarlas. He says he has looked through most and will finish tomorrow.

**Review by DavidKarlas — APPROVED** [review]
Sorry for slow approval.

**Comment by vcschapp** [comment]
Merging to `dev` per review approval status and alignment in this week's Schema TF around "pure docs" PRs.

**Pull request merged**